sdk/cognitivelanguage/ai-language-text/src/lro.ts

```diff
@@ -47,19 +47,36 @@ const jobStatusOperationSpec: OperationSpec = {
   serializer,
 };
 
+function addOnResponse<TOptions extends OperationOptions>(
+  options: TOptions,
+  cb: (rawResponse: FullOperationResponse, response: unknown, error: unknown) => void
+): TOptions {
+  return {
+    ...options,
+    onResponse: (rawResponse, response, error) => {
+      cb(rawResponse, response, error);
+      options.onResponse?.(rawResponse, response, error);
+    },
+  };
+}
+
+function logWanrHeader(rawResponse: FullOperationResponse) {
+  const warnHeader = rawResponse.headers.get("warn-text");
+  if (warnHeader) {
+    warnHeader.split(";").map(logger.warning);
```

Comment:
Just wondering, does this by default get logged or does it only get logged if a user opts-in to logging?

I ask because right now Python does the latter. And I'm wondering if this is something we want to surface regardless of whether logging is enabled.

Reply:
It is opt-in.

>  I'm wondering if this is something we want to surface regardless of whether logging is enabled.

IIRC, we discussed this with Mikael a while back and he seemed fine with the logging approach. We can add them to our response objects later if there is a customer ask.

Reply:
Okay, good to know the service team is good with it.